src/routes/ProtectedRoute/index.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Navigate, Outlet } from 'react-router-dom';
+
+import { useAppSelector } from 'hooks';
+import { paths } from 'routes';
+
+// Tutorial Link:
+// https://blog.logrocket.com/complete-guide-authentication-with-react-router-v6/
+// https://blog.logrocket.com/handling-user-authentication-redux-toolkit/
+const ProtectedRoute = (): JSX.Element => {
+  const { token } = useAppSelector((state) => state.auth);
```

Comment:
Is it always having `token` (regardless the value) in the `state.auth` object? It might crash the app if `state.auth` returns undefined 🤔 Be careful with that

Reply:
It seems it was added in the initialState [here](https://github.com/manh-t/react-survey/pull/56/files#diff-445a58788e0d28fcf32135554f18f6bcf0b7d194e74cdf6edd7d0855073e5e94R21)

Reply:
@nvminhtue don't worry, we always have an initial state 🙋